### ISSUE TITLE
Documentation is not synced with library 

### ISSUE THREAD

**Issue by adisve** [realm/realm-swift#8262]
### How frequently does the bug occur?

Always

### Description

The documentation for how to use Actor-Isolated Realms gives a clear basic example:

```swift
@MainActor
func mainThreadFunction() async throws {
// These are identical: the async init produces a
// MainActor-isolated Realm if no actor is supplied
let realm1 = try await Realm()
let realm2 = try await Realm(actor: MainActor.shared)
 
try await useTheRealm(realm: realm1)
}
```

But attempting to create a Realm instance in this way using the latest Realm release gives the error 

```Extra argument 'actor' in call```

for this line:

```let realm2 = try await Realm(actor: MainActor.shared)```

or even this line, when attempting to define your own actor-isolated realm 

```realm = try await Realm(actor: self)```

### Can you reproduce the bug?

Always

### Reproduction Steps

Copy example code from https://www.mongodb.com/docs/realm/sdk/swift/actor-isolated-realm/#define-a-custom-realm-actor and attempt to use it in a project with the latest Realm release (10.40.1)

### Version

10.40.1

### What Atlas Services are you using?

Local Database only

### Are you using encryption?

No

### Platform OS and version(s)

macOS Monterey 12.6.6

### Build environment

Xcode version: 14.2
Dependency manager and version: Swift Package Manager

**Comment by dianaafanador3** [comment]
@adisve this feature is available in Code versions which have Swift v5.8 or greater. Unfortunately this is available only from Xcode 14.3 version or later. This is mentioned on the documentation on the [prerequisites](https://www.mongodb.com/docs/realm/sdk/swift/actor-isolated-realm/#prerequisites).

**Comment by adisve** [comment]
> @adisve this feature is available in Code versions which have Swift v5.8 or greater. Unfortunately this is available only from Xcode 14.3 version or later. This is mentioned in the documentation on the [prerequisites](https://www.mongodb.com/docs/realm/sdk/swift/actor-isolated-realm/#prerequisites).

I see, looks like I missed this. Thanks for the clarification!
